beam-migrate/Database/Beam/Migrate/Backend.hs

```diff
@@ -88,9 +88,16 @@ data BeamMigrationBackend be m where
     , backendFileExtension :: String
     , backendConvertToHaskell :: HaskellPredicateConverter
     , backendActionProvider :: ActionProvider be
-    , backendTransact :: forall a. String -> m a -> IO (Either DdlError a)
+    , backendRunSqlScript :: Text -> m ()
```

Comment:
Writing SQL manually can lead to errors.
How about having `backendRunSqlScript :: Text -> m (Either (SqlException be) ())` or something?

For example, `postgresql-simple`'s `execute` can [throw exceptions](https://hackage.haskell.org/package/postgresql-simple-0.7.0.0/docs/Database-PostgreSQL-Simple.html#v:execute)

Reply:
Regarding exceptions: all these monad actions run in `m` (which is an existential type here) which can only be executed by `backendRun` in `BeamMigrateConnection`, which returns an `Either` type. I agree that `DdlError` (thin wrapper over a string) is not the right type, but the main concern that exceptions could be thrown is handled by the fact that all these actions need to be run by this potentially-failable runner.

Reply:
And then for SQL scripts... I've tried a variety of ways of doing this, and for DDL at least, I think it's necessary to allow fallbacks to SQL scripts. There's too many databases, and options to realistically support this as a DSL. We already cover the basic DDL languages via beam-migrate.